src/commands/deleteResourceGroup/v2/deleteResourceGroupV2.ts

```diff
@@ -3,72 +3,110 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { AzureWizard, IActionContext, UserCancelledError } from '@microsoft/vscode-azext-utils';
+import { ResourceGroup } from '@azure/arm-resources';
+import { uiUtils } from '@microsoft/vscode-azext-azureutils';
+import { AzureWizard, IActionContext, IAzureQuickPickItem, nonNullProp, subscriptionExperience, UserCancelledError } from '@microsoft/vscode-azext-utils';
+import { AzureResource, AzureSubscription } from '@microsoft/vscode-azext-utils/hostapi.v2';
+import { DefaultAzureResourceProvider } from '../../../api/DefaultAzureResourceProvider';
 import { ext } from '../../../extensionVariables';
 import { GroupingItem } from '../../../tree/azure/GroupingItem';
 import { createActivityContext } from '../../../utils/activityUtils';
+import { createResourceClient } from '../../../utils/azureClients';
 import { localize } from '../../../utils/localize';
 import { settingUtils } from '../../../utils/settingUtils';
+import { createSubscriptionContext } from '../../../utils/v2/credentialsUtils';
 import { DeleteResourceGroupContext } from '../DeleteResourceGroupContext';
 import { DeleteResourceGroupStep } from '../DeleteResourceGroupStep';
 
 export async function deleteResourceGroupV2(context: IActionContext, primaryNode?: GroupingItem, selectedNodes?: GroupingItem[]): Promise<void> {
+
+    // unset nodes that are not resource groups
+    selectedNodes = selectedNodes?.filter(n => !!n.resourceGroup);
+    if (!primaryNode?.resourceGroup) {
+        primaryNode = undefined;
+    }
+
+    let subscription: AzureSubscription;
+    let resourceGroupsToDelete: AzureResource[] = [];
+
     if (!selectedNodes) {
```

Comment:
This is a nightmare. I think we may be able to have a wrapper to handle this logic for all of our functions at some point.

Reply:
Agreed, the whole selected vs primary is pretty confusing.

Reply:
I've added a utility to make it not as ugly.

Reply:
Docker has a similar challenge: https://github.com/microsoft/vscode-docker/blob/main/src/utils/multiSelectNodes.ts

When multiple nodes are selected and an action performed on one of them, we simply ignore those for which the action is not valid.

Reply:
Thanks Brandon, I'll file an issue on azuretools to create a shared utility for this stuff. Hopefully we can come up with a few utils that cover everyone's use case.